t/subroutines.t

```diff
@@ -0,0 +1,24 @@
+use Test::More;
+
+use Data::Checks;
+use lib qw< tlib t/tlib >;
+use Data::Checks::TestUtils 'UINT';
+
+sub foo ( $max_size :of(UINT) ) {
+    $max_size = -2.43;
+}
+
+FAIL_ON_PARAM { foo(-42) } 'foo(-42)';
+
+{
+eval { foo(3) };
+my $err = $@ // '';
+like( $err, qr/\QCan't assign -2.43 to \E\$max_size: failed UINT check/, 'foo(3)' );
+like( $err, qr{\Qat t/subroutines.t line 8\E}, '...at correct location' );
```

Comment:
Note: (self promoting `Test::YAFT`)
```
it q (should report failed UINT <- FLOAT assignment (at correct location))
  => got    { foo (3) }
  => throws =>
    & expect_re (qr ( ... Can't  ... ))
    & expect_re (qr ( ... at line ... ))
;
```
`it` will also report `expected to die lives` (similar to `throws_ok` but accepts `Test::Deep` expectations)